### DISCUSSION TITLE
Do we need to Normalize the adj matrix for other models except GCN?

### DISCUSSION THREAD

**Discussion by monk1337** [pyg-team/pytorch_geometric#3108] [Q&A]
Hi, I am using this awesome package and have one doubt,
I am trying to use models other than GCN and noticed that only GCN is performing normalization internally :

```
D = torch.pow(A.sum(1).float(), -0.5)
D = torch.diag(D)
adj = torch.matmul(torch.matmul(A, D).t(), D)
```

I wanted to know is this normalization only necessary for GCN or we need to perform manual transformation for other models such as GAT, GRAPHsage, GIN etc

what is the performance effect?

**Comment by rusty1s** [comment]
In general, you do not have to care about normalization as PyG will normalize according to the GNN formula automatically. For example, for `GraphSAGE`, we use `mean` normalization. On the other-hand, `GIN` is defined to not do any normalization, so it will perform `sum` aggregation.
